apps/evm_family/evm_contracts.c

```diff
@@ -281,7 +281,13 @@ uint8_t ETH_ExtractArguments(const uint8_t *pAbiPayload,
   const uint8_t *pPayloadBasePtr = pCurrHeadPtr;
   uint8_t currArgument;
 
-  for (currArgument = 0; currArgument < numArgsInFunction; currArgument++) {
+  for (currArgument = 0; currArgument <= numArgsInFunction; currArgument++) {
+    // only return OK when completely parsed whole payload
```

Comment:
The comment can mention the case where no arguments and/or why it is valid.

Reply:
updated